backend/typescript/services/implementations/adminService.ts

```diff
@@ -0,0 +1,134 @@
+import prisma from "../../prisma";
+import type {
+  IAdminService,
+  NotificationDTO,
+} from "../interfaces/adminService";
+// import type ResidentDTO from "../interfaces/FILEOFDTO";
+// import type IStaffService from "../interfaces/FILEOFDTO";
+import logger from "../../utilities/logger";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import { IResidentService } from "../interfaces/residentService";
+import ResidentService from "./residentService";
+
+const residentService: IResidentService = new ResidentService();
+const Logger = logger(__filename);
+
+class AdminService implements IAdminService {
+  staffId: number;
+
+  constructor(staffId: number) {
+    this.staffId = staffId;
+  }
+
+  async getAllNotifications(): Promise<NotificationDTO[]> {
+    try {
+      const notifications = await prisma.notification.findMany({
+        include: { residents: true },
+      });
+      if (!notifications) throw new Error(`No residents found.`);
+      return notifications;
+    } catch (error) {
+      Logger.error(
+        `Failed to get all active Residents. Readon = ${getErrorMessage(
+          error,
+        )}`,
+      );
+      throw error;
+    }
+  }
+
+  async getNotificationById(id: number): Promise<NotificationDTO> {
+    try {
+      const notification = await prisma.notification.findUnique({
+        where: {
+          id: Number(id),
+        },
+        include: {
+          residents: true,
+        },
+      });
+      if (!notification) throw new Error(`notification id ${id} not found`);
+
+      return notification;
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to get Notification. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+  }
+
+  async sendNotification(
+    notif_message: string,
+    resident_id: number,
+  ): Promise<NotificationDTO> {
+    let newNotification: NotificationDTO;
+    try {
+      newNotification = await prisma.notification.create({
+        data: {
+          message: String(notif_message),
```

Comment:
Correct me if I'm wrong, but I don't think the String(...) function does anything here as the notif_message should already be a string going in. (the number case is different cause graph ql id's are treated as strings from apollo servers).

Comment also applicable for Line 106